### PR TITLE
Add new guide in C4R for Query parameters

### PR DESCRIPTION
**Pull request by Josmorsot**

### CODE REVIEW

File: app/content/react/guides/query-parameters.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+In this guide we're going to create a simple map that shows the earthquakes in Spain and allows to filter it by their magnitude.
+
+#### Creating a map using QueryParameters
+
+First you'll need to create a source that supports query parameters. In this case we are getting all the earthquakes whoses magnitude are between `min: 4` and `max: 5`.
+
+```ts
+import { MAP_TYPES } from '@deck.gl/carto';
+
+const sql = 'select * from `carto-demo-data.demo_tables.spain_earthquakes` where magnitude @min and @max';
```
Comment (VictorVelarde) [inline app/content/react/guides/query-parameters.md:13]: This query lacks BETWEEN, right?

File: app/content/react/guides/query-parameters.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  data: sql,
+  type: MAP_TYPES.QUERY,
+  queryParameters: {
+    min: 4,
+    max: 5
+  }
+}
+```
+
+Once you have your source is ready to use you can create your layer using `useCartoLayerProps` hook.
```
Comment (VictorVelarde) [inline app/content/react/guides/query-parameters.md:26]:
```suggestion
Once you have your source ready to be used, you can create your layer using `useCartoLayerProps` hook.
```

File: app/content/react/guides/query-parameters.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
+## Query Parameters
+
+You can use QueryParameters to filter your layers at backend side, you can check our [documentation]({{< ref "/deck-gl/reference#queryparameters-depends-on-provider-optional" >}} "About Us") for different providers.
```
Comment (VictorVelarde) [inline app/content/react/guides/query-parameters.md:3]:
```suggestion
You can use QueryParameters to filter your layers at backend side, using the concept of *parametrized queries*. You can check our related [deck.gl documentation]({{< ref "/deck-gl/reference#queryparameters-depends-on-provider-optional" >}} "About Us") for different providers.
```

File: app/content/react/guides/query-parameters.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  ...layerProps,
+  id: 'earthquakesLayer',
+  pointRadiusUnits: 'pixels',
+  lineWidthUnits: 'pixels',
+  getLineColor: [0, 0, 0],
+  getPointRadius: 3,
+});
+```
+
+Your layer is ready and you can use it your map, now we're going to create a `<FormulaWidget />` wich display the number of earthquakes in your viewport.
```
Comment (VictorVelarde) [inline app/content/react/guides/query-parameters.md:43]:
```suggestion
Now your layer is ready and you can use it in your map, so we're going to create a `<FormulaWidget />` which displays the number of earthquakes in your viewport.
```

File: app/content/react/guides/query-parameters.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+#### Using queryParameters with widgets
+All our widgets are ready to use queryParameters. You just need to pass the datasource properly configured.
```
Comment (VictorVelarde) [inline app/content/react/guides/query-parameters.md:46]:
```suggestion
All our widgets are ready to use queryParameters. You just need to pass the `datasource` properly configured (that basically means declaring its `type` as MAP_TYPES.QUERY and adding the corresponding `queryParameters` configuration). After that, you can use any widget as usual, eg:

File: app/content/react/guides/query-parameters.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+<FormulaWidget
+  id='totalEarthquakes'
+  title='Total earthquakes'
+  dataSource={earthquakesSource.id}
+  operation={AggregationTypes.COUNT}
+  />
+```
+
+#### Filtering with queryParameters using a widget
+We can use widgets to modify and apply filters using queryParameters. We're going to add a `<RangeWidgetUI />` to allows filtering by magnitude. First we need to modify the source to support modify `queryParameters`.
```
Comment (VictorVelarde) [inline app/content/react/guides/query-parameters.md:58]:
```suggestion
But the utility of queryParameters is dynamically changing the parametrized values in the app, so now we're gonna use a widget to modify those values (and as a consequence, to apply new backend filters using queryParameters). Here we're going to add a `<RangeWidgetUI />` to allow filtering earthquakes by magnitude. 

First we need to prepare the source to support `queryParameters` modification, passing those values as parameters to it.
```

File: app/content/react/guides/query-parameters.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+In this guide we're going to create a simple map that shows the earthquakes in Spain and allows to filter it by their magnitude.
+
+#### Creating a map using QueryParameters
+
+First you'll need to create a source that supports query parameters. In this case we are getting all the earthquakes whoses magnitude are between `min: 4` and `max: 5`.
+
+```ts
+import { MAP_TYPES } from '@deck.gl/carto';
+
+const sql = 'select * from `carto-demo-data.demo_tables.spain_earthquakes` where magnitude between @min and @max';
```
Comment (padawannn) [inline app/content/react/guides/query-parameters.md:13]: This syntax (parametrized queries) is only compatible with BigQuery. I think that we should specify it. What about mentioning how would be the same query with another provider

File: app/content/react/guides/query-parameters.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+  lineWidthUnits: 'pixels',
+  getLineColor: [0, 0, 0],
+  getPointRadius: 3,
+});
+```
+
+Now your layer is ready and you can use it in your map, so we're going to create a `<FormulaWidget />` which displays the number of earthquakes in your viewport.
+
+#### Using queryParameters with widgets
+All our widgets are ready to use queryParameters. You just need to pass the `datasource` properly configured (that basically means declaring its `type` as MAP_TYPES.QUERY and adding the corresponding `queryParameters` configuration). After that, you can use any widget as usual, eg:
```
Comment (padawannn) [inline app/content/react/guides/query-parameters.md:46]: This sentence `All our widgets are ready to use queryParameters` sounds confusing to me. Probably I would say
All our widgets are ready to use **datasources** with queryParameters

File: app/content/react/guides/query-parameters.md (hunk 7 above)
Comment (Josmorsot) [reply app/content/react/guides/query-parameters.md:13]: it's just an example. I've added a link to the documentation for queryParameters, where different providers are specified. Anyways I think that would be great to specify that this example is based on BQ

File: app/content/react/guides/query-parameters.md (hunk 8 above)
Comment (Josmorsot) [reply app/content/react/guides/query-parameters.md:46]: Much better

File: app/content/react/guides/query-parameters.md (hunk 7 above)
Comment (eamador) [reply app/content/react/guides/query-parameters.md:13]: I agree with @Josmorsot it'd be enough to point that this example is for BQ and then link to the `queryParameters` documentation with the specifics for each provider.

**Review by eamador — CHANGES_REQUESTED** [review]
I think it's nicely documented! Just keep in mind @padawannn's comments

**Review by VictorVelarde — APPROVED** [review]
LGTM Iván. Let's have a general validation from Moisés, and we'll land it later

File: app/content/react/guides/query-parameters.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
+## Query Parameters
+
+You can use QueryParameters to filter your layers & widgets at the backend side, using the concept of *parametrized queries* over your sources. You can check our related [deck.gl documentation]({{< ref "/deck-gl/reference#queryparameters-depends-on-provider-optional" >}} "About Us") for different providers.
```
Comment (VictorVelarde) [inline app/content/react/guides/query-parameters.md:3]: @Josmorsot This "About Us" looks wrong. Maybe a copy & paste. If we keep it, we should display something like "Query parameters in deck.gl"

**Review by moimart — APPROVED** [review]

File: app/content/react/guides/query-parameters.md (hunk 9 above)
Comment (VictorVelarde) [inline app/content/react/guides/query-parameters.md:3]:
```suggestion
You can use QueryParameters to filter your layers & widgets at the backend side, using the concept of *parametrized queries* over your sources. You can check our related [deck.gl documentation]({{< ref "/deck-gl/reference#queryparameters-depends-on-provider-optional" >}} "queryParameters in CartoLayer") for different providers.
```

**Pull request merged**